dust-lambda-extension/extensions/dust

```diff
@@ -0,0 +1,294 @@
+
+# Global variables
+EXTENSION_ID=""
+EXIT_REQUESTED=false
+
+# Improved error handling for curl requests
+make_api_request() {
+	local method="$1"
+	local url="$2"
+	local headers=("${@:3}")
+
+	local temp_file
+	temp_file=$(mktemp)
+	local http_code
+
+	# Use array to properly handle headers with spaces
+	local curl_args=(-s -w "%{http_code}" -o "$temp_file" -X "$method" "$url")
+	for header in "${headers[@]}"; do
+		curl_args+=(-H "$header")
+	done
+
+	http_code=$(curl "${curl_args[@]}")
+	local curl_exit_code=$?
+
+	if [[ $curl_exit_code -ne 0 ]]; then
+		LOG_ERROR "curl failed with exit code $curl_exit_code for $method $url"
+		rm -f "$temp_file"
+		return 1
+	fi
+
+	if [[ $http_code -lt 200 || $http_code -ge 300 ]]; then
+		LOG_ERROR "HTTP request failed with status $http_code for $method $url"
+		LOG_DEBUG "Response: $(cat "$temp_file")"
+		rm -f "$temp_file"
+		return 1
+	fi
+
+	cat "$temp_file"
+	rm -f "$temp_file"
+	return 0
+}
+
+# Handle signals cleanly with proper cleanup
+handle_signal() {
+	LOG_INFO "Received termination signal. Preparing to shut down."
+	EXIT_REQUESTED=true
+	do_shutdown
+}
+
+do_shutdown() {
+	LOG_INFO "Starting clean shutdown process."
+
+	if [[ -n $EXTENSION_ID && $EXIT_REQUESTED == "true" ]]; then
+		LOG_DEBUG "Signaling extension shutdown to Lambda."
+		# Signal the Lambda platform that we are exiting
+		if ! make_api_request "POST" "$EXIT_API" "Lambda-Extension-Identifier: $EXTENSION_ID" >/dev/null 2>&1; then
+			LOG_WARNING "Failed to notify Lambda platform of extension shutdown"
+		fi
+	fi
+
+	# Clean up temporary files
+	[[ -f $TMP_CHALK_PATH ]] && rm -f "$TMP_CHALK_PATH"
+
+	exit 0
+}
+
+# Setup binary with proper validation and error handling
+setup_chalk_binary() {
+	# Check if chalk binary exists and is executable
+	if [[ ! -x $CHALK_BIN_PATH ]]; then
+		LOG_ERROR "chalk binary not found or not executable at $CHALK_BIN_PATH"
+		return 1
+	fi
+
+	# Copy chalk binary to /tmp if it doesn't exist or is outdated
+	if [[ ! -x $TMP_CHALK_PATH ]] || [[ $CHALK_BIN_PATH -nt $TMP_CHALK_PATH ]]; then
+		LOG_INFO "Copying chalk binary to /tmp"
+		if ! cp "$CHALK_BIN_PATH" "$TMP_CHALK_PATH"; then
+			LOG_ERROR "Failed to copy chalk binary to /tmp"
+			return 1
+		fi
+
+		if ! chmod +x "$TMP_CHALK_PATH"; then
+			LOG_ERROR "Failed to make chalk binary executable in /tmp"
+			return 1
+		fi
+	fi
+
+	return 0
+}
+
+# Process chalk marks with better error handling
+process_chalk_marks() {
+	LOG_INFO "$TMP_CHALK_PATH extracting $CHALK_MARK_PATH"
+	if [[ -f $CHALK_MARK_PATH ]]; then
+		LOG_DEBUG "Found $CHALK_MARK_PATH, running chalk extract."
+		if ! "$TMP_CHALK_PATH" extract "$CHALK_MARK_PATH"; then
+			LOG_WARNING "chalk extract command failed"
+			return 1
+		fi
+	else
+		LOG_WARNING "No chalk.json found at $CHALK_MARK_PATH"
+		return 1
+	fi
+
+	return 0
+}
+
+# JSON parsing using pure bash/standard tools (no jq dependency)
+extract_json_field() {
```

Comment:
lambda does not send these as return headers?

makes for parsing responses much much better

Reply:
It returns the event UUID but not the event type